packages/calypso-products/src/get-plan-term-variant.ts

```diff
@@ -0,0 +1,26 @@
+import { TERMS_LIST } from './constants/terms';
+import { PLANS_LIST } from './plans-list';
+import type { PlanSlug } from './types';
+
+/**
+ * Return a (monthly/yealry/2yearly/etc) variant of a plan
+ *
+ * @param {PlanSlug} slug Slug/name of the plan to get the yearly variant from
+ * @param {typeof TERMS_LIST[ number ]} planTerm The term to find a variant for
+ * @returns {PlanSlug} Slug of the term-variant plan
+ */
+export function getPlanSlugForTermVariant(
```

Comment:
We have a few (incomplete) ways to do similar right now e.g. per term selectors `getYearlyFromMonthly` (etc.), or even `getAnnualSlug` props on monthly product definitions in the product lists). I think a canonical/polymorphic form should serve things better for any sort of plan conversion that may come up. If we leave this out there, things may gradually converge here (rather than continue with the old). 

There's a similar utility for products ([getProductYearlyVariant](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-products/src/get-product-yearly-variant.ts)) but also built a little more restrictively.

Happy with either/or. If we keep this, we can add some tests to guard.

Reply:
The introduction of 2-yearly and 3-yearly plans basically means we need to generalize the code regarding the term eventually, and this looks like a great start :) 

We can surely add test coverage as a separate PR to keep the scope of this PR.

Reply:
> We can surely add test coverage as a separate PR to keep the scope of this PR.

Yes, let's do that to keep things moving. 🙌 We can come back to this later this/next week.